test/suites/expiration/suite_test.go

```diff
@@ -66,6 +68,310 @@ var _ = AfterEach(func() { env.Cleanup() })
+			env.EventuallyExpectHealthyPodCount(selector, int(numPods))
+			env.Monitor.Reset() // Reset the monitor so that we can expect a single node to be spun up after expiration
+
+			By("scaling down the deployment")
+			// Update the deployment to a third of the replicas.
+			dep.Spec.Replicas = lo.ToPtr[int32](3)
+			env.ExpectUpdated(dep)
+
+			By("spreading the pods to each of the nodes")
+			env.EventuallyExpectHealthyPodCount(selector, 3)
+			var nodes []*v1.Node
+			// Delete pods from the deployment until each node has one pod.
+			nodePods := []*v1.Pod{}
+			for {
+				nodes = env.EventuallyExpectNodeCount("==", 3)
+				node, found := lo.Find(nodes, func(n *v1.Node) bool {
+					nodePods = env.ExpectHealthyPodsForNode(n.Name)
+					return len(nodePods) > 1
+				})
+				if !found {
+					break
+				}
+				// Set the nodes to unschedulable so that the pods won't reschedule.
+				node.Spec.Unschedulable = true
+				env.ExpectUpdated(node)
+				for _, pod := range nodePods[1:] {
+					env.ExpectDeleted(pod)
+				}
+				Eventually(func(g Gomega) {
+					g.Expect(len(env.ExpectHealthyPodsForNode(node.Name))).To(Equal(1))
+				}).WithTimeout(5 * time.Second).Should(Succeed())
+			}
+			env.EventuallyExpectHealthyPodCount(selector, 3)
+
+			By("cordoning and adding finalizer to the nodes")
+			nodes = env.EventuallyExpectNodeCount("==", 3)
+			// Add a finalizer to each node so that we can stop termination disruptions
+			for _, node := range nodes {
+				node.Finalizers = append(node.Finalizers, common.TestingFinalizer)
+				// Set nodes as unschedulable so that pod nomination doesn't delay disruption for the second disruption action
+				node.Spec.Unschedulable = true
+				env.ExpectUpdated(node)
+			}
+
+			By("expiring the nodes")
+			// expire the nodeclaims
+			nodePool.Spec.Disruption.ExpireAfter = corev1beta1.NillableDuration{Duration: lo.ToPtr(time.Second * 30)}
+			env.ExpectUpdated(nodePool)
+
+			// Expect that the NodeClaims will all be marked expired
+			Eventually(func(g Gomega) {
+				nodeClaimList := &corev1beta1.NodeClaimList{}
+				err := env.Client.List(env.Context, nodeClaimList)
+				g.Expect(err).To(Succeed())
+				lo.ForEach(nodeClaimList.Items, func(nc corev1beta1.NodeClaim, _ int) {
+					g.Expect(nc.StatusConditions().GetCondition(corev1beta1.Expired).IsTrue()).To(BeTrue())
+				})
+			}).Should(Succeed())
+
+			By("enabling disruption by removing the do not disrupt annotation")
+			pods := env.EventuallyExpectHealthyPodCount(selector, 3)
+			// Remove the do-not-disrupt annotation so that the nodes are now disruptable
+			for _, pod := range pods {
+				delete(pod.Annotations, corev1beta1.DoNotDisruptAnnotationKey)
+				env.ExpectUpdated(pod)
+			}
+
+			// List nodes so that we get any updated information on the nodes. If we don't
+			// we have the potential to over-write any changes Karpenter makes to the nodes.
+			nodes = env.EventuallyExpectNodeCount("==", 3)
+
+			nodes[0].Spec.Unschedulable = false
```

Comment:
What's the purpose in marking these as schedulable again? Looks like it's to revert marking them as unschedulable while spreading the pods but do they need to be schedulable for expiration to work?

Reply:
We mark them schedulable so that the pods on the other nodes can schedule to these nodes, which will result in two delete operations. Although, I'm realizing that I only need to mark one node as schedulable, as the other two will be able to delete.